spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java

```diff
@@ -307,6 +341,17 @@ private FetchResult fetch(Git git, String label) {
 			}
 			return result;
 		}
+		catch (TransportException ex){
+			if(errorLogOnFailingFetch){
```

Comment:
But I'm saying that we try for branches that don't have to exist, therefore aren't necessarily errors.